server/index.js

```diff
@@ -0,0 +1,38 @@
+const express = require("express");
+const app = express();
+
+const http = require("http");
+const cors = require("cors");
+const { Server } = require("socket.io");
+app.use(cors());
+const server = http.createServer(app);
+
+const io = new Server(server, {
+  cors: {
+    origin: "http://localhost:3000",
+    methods: ["GET", "POST"],
+  },
+});
+
+io.on("connection", (socket) => {
+  console.log(`User Connected ${socket.id}`);
+
+  socket.on("join_room", (data) => {
+    socket.join(data);
+    console.log(`User with ID ${socket.id} joined room: ${data}`);
+  });
+
+  socket.on("send_message", (data) => {
+    console.log(data);
+    socket.to(data.room).emit("receive_message", data);
+  });
+
+  //Disconnect socket when React App refreshes
+  socket.on("disconnect", () => {
+    console.log("connection disconnected", socket.id);
+  });
+});
+
+server.listen(3001, () => {
+  console.log("Server listening on port 3001");
```

Comment:
add a utility for logging for such cases.
Other than that, we don't use console logs directly

Reply:
Noted and Resolved